src/core/uint.js

```diff
@@ -166,7 +186,7 @@ UInt.prototype.parse_generic = function(j) {
 };
 
 UInt.prototype.parse_hex = function(j) {
-  if (typeof j === 'string' && j.length === (this.constructor.width * 2)) {
+  if (new RegExp(`^[0-9A-F]{${this.constructor.width * 2}}$`).test(j)) {
```

Comment:
The idea here is that one can rely on `uint*.is_valid()` to check that input will be reflected by rippled. I think it's a bad idea to allow setting an `InvoiceID` in lowercase that will be returned in uppercase, e.g., and I think `UInt256.is_valid()` is a good place to validate.